### PR TITLE
HWS-196: Remove recipes base and add snoop compiler

### PR DESCRIPTION
**Pull request by gita135**
Removed deprecated RecipesBase library and added in plots.jl and snoopPrecompiler.jl

### CODE REVIEW

**Comment by mofii** [comment]
Hi @gita135,
I just tried using `RecipesBase` and it worked. I'm therefore not currently convinced that we need to remove `RecipesBase` and include `Plots`. Can you convince me?
See this example from their Redme.md:

```julia
(@v1.8) pkg> activate --temp
  Activating new project at `/var/folders/2h/vd2kqpbj1xq8c6dyy2n7ps8w0000gn/T/jl_UxIRt5`

(jl_UxIRt5) pkg> add RecipesBase
   Resolving package versions...
    Updating `/private/var/folders/2h/vd2kqpbj1xq8c6dyy2n7ps8w0000gn/T/jl_UxIRt5/Project.toml`
  [3cdcf5f2] + RecipesBase v1.3.2
    Updating `/private/var/folders/2h/vd2kqpbj1xq8c6dyy2n7ps8w0000gn/T/jl_UxIRt5/Manifest.toml`
  [3cdcf5f2] + RecipesBase v1.3.2
  [66db9d55] + SnoopPrecompile v1.0.1

julia> using RecipesBase

julia> # Our user-defined data type
       struct T end

julia> @recipe function plot(::T, n = 1; customcolor = :green)
           markershape --> :auto        # if markershape is unset, make it :auto
           markercolor :=  customcolor  # force markercolor to be customcolor
           xrotation   --> 45           # if xrotation is unset, make it 45
           zrotation   --> 90           # if zrotation is unset, make it 90
           rand(10,n)                   # return the arguments (input data) for the next recipe
       end

julia> using Plots

julia> gr()
Plots.GRBackend()

julia> plot(T(), 5; customcolor = :black, shape=:c)
```

File: src/TcpInstruments.jl [hunk 1]
```diff
@@ -46,7 +46,10 @@ using Base.Threads: @spawn
 using Dates
 using MAT
 using JLD2
-using RecipesBase
+using SnoopPrecompile
+@precompile_setup begin
+    using Plots
```
Comment (mofii) [inline src/TcpInstruments.jl:51]: This is not enough for SnoopPreCompile to work. You need to run your actual use cases for this package to work.

**Review by mofii — CHANGES_REQUESTED** [review]
The `SnoopPreCompile` package isn't recording any commands to precompile which is most likely why it's not resulting in any speedup.
See also the other comment of me not being convinced that we need to load Plots.jl.

**Comment by gita135** [comment]
> Hi @gita135, I just tried using `RecipesBase` and it worked. I'm therefore not currently convinced that we need to remove `RecipesBase` and include `Plots`. Can you convince me? See this example from their Readme.md:
> 
> ```julia
> (@v1.8) pkg> activate --temp
>   Activating new project at `/var/folders/2h/vd2kqpbj1xq8c6dyy2n7ps8w0000gn/T/jl_UxIRt5`
> 
> (jl_UxIRt5) pkg> add RecipesBase
>    Resolving package versions...
>     Updating `/private/var/folders/2h/vd2kqpbj1xq8c6dyy2n7ps8w0000gn/T/jl_UxIRt5/Project.toml`
>   [3cdcf5f2] + RecipesBase v1.3.2
>     Updating `/private/var/folders/2h/vd2kqpbj1xq8c6dyy2n7ps8w0000gn/T/jl_UxIRt5/Manifest.toml`
>   [3cdcf5f2] + RecipesBase v1.3.2
>   [66db9d55] + SnoopPrecompile v1.0.1
> 
> julia> using RecipesBase
> 
> julia> # Our user-defined data type
>        struct T end
> 
> julia> @recipe function plot(::T, n = 1; customcolor = :green)
>            markershape --> :auto        # if markershape is unset, make it :auto
>            markercolor :=  customcolor  # force markercolor to be customcolor
>            xrotation   --> 45           # if xrotation is unset, make it 45
>            zrotation   --> 90           # if zrotation is unset, make it 90
>            rand(10,n)                   # return the arguments (input data) for the next recipe
>        end
> 
> julia> using Plots
> 
> julia> gr()
> Plots.GRBackend()
> 
> julia> plot(T(), 5; customcolor = :black, shape=:c)
> ```

@mofii  Yes the ```RecipesBase.jl``` lib still works, but it is deprecated and was migrated to ```Plots.jl``` library. From your comment on PR #137 I understood that bc of that, you wanted it removed? I cannot use ```@recipe``` without either ```RecipesBase.jl``` or ``` Plots.jl``` let me know which you prefer

**Comment by mofii** [comment]
> @mofii Yes the `RecipesBase.jl` lib still works, but it is deprecated and was migrated to `Plots.jl` library. From your comment on PR #137 I understood that bc of that, you wanted it removed? I cannot use `@recipe` without either `RecipesBase.jl` or ` Plots.jl` let me know which you prefer

@gita135 I'm not actually seeing which comment you are referring to in that PR?
I agree, though, that we had decided to remove RecipesBase, based on the information that it's been deprecated. But as I show above, I don't see any signs that it's been deprecated? I thought there would be a warning that the package has been deprecated. 
Taking a closer look, the package has just been moved to a different repository. Moving the package to a different repository is not the same as it being deprecated. I don't know why the developers chose to move the RecipesBase package to the Plots.jl repository, but it doesn't seem to change anything for us? 
In the example I show that I have version 1.3.2 installed (which was released today), and the latest release from the original repository is version 1.3.0. In other words, the package is still actively maintained.

**Comment by gita135** [comment]
> > @mofii Yes the `RecipesBase.jl` lib still works, but it is deprecated and was migrated to `Plots.jl` library. From your comment on PR #137 I understood that bc of that, you wanted it removed? I cannot use `@recipe` without either `RecipesBase.jl` or ` Plots.jl` let me know which you prefer
> 
> @gita135 I'm not actually seeing which comment you are referring to in that PR? I agree, though, that we had decided to remove RecipesBase, based on the information that it's been deprecated. But as I show above, I don't see any signs that it's been deprecated? I thought there would be a warning that the package has been deprecated. Taking a closer look, the package has just been moved to a different repository. Moving the package to a different repository is not the same as it being deprecated. I don't know why the developers chose to move the RecipesBase package to the Plots.jl repository, but it doesn't seem to change anything for us? In the example I show that I have version 1.3.2 installed (which was released today), and the latest release from the original repository is version 1.3.0. In other words, the package is still actively maintained.

Ahh, I see, thank you. The moving of the pkg did not make sense to me and I assumed it was deprecated. Happy to know we can continue to use the RecipesBase without loading Plots. I can close the PRs pertaining to removing RecipesBase.

**Comment by mofii** [comment]
closing

**Pull request closed without merging**